File: Part 2 - Simple_Linear_Regression_sample/simple_linear_regression.r
#**Import the data set**
setwd("C:/Pradeep/Project/Learning/ML A-Z/P1 Data Preprocessing/Machine-Learning-A-Z/Part 2 - Regression/Section 4 - Simple Linear Regression/Simple_Linear_Regression_sample/")
real_dataset=read.csv("Salary_Data.csv")

# Splitting the dataset into the Training set and Test set
# install.packages('caTools')
library(caTools)
set.seed(123)
real_split = sample.split(real_dataset$Salary, SplitRatio = 2/3)
real_training_set = subset(real_dataset, real_split == TRUE)
real_test_set = subset(real_dataset, real_split == FALSE)


# Fitting Simple Linear Regression to the Training Set
real_regressor=lm(formula = Salary ~ YearsExperience,data = real_training_set)
summary(real_regressor)

# Predicting the test results
real_y_pred=predict(real_regressor,newdata=real_test_set)

#Visualizing training test results
#install.packages('ggplot2')
library(ggplot2)
ggplot() + 
  geom_point(aes(x=real_training_set$YearsExperience,y = real_training_set$Salary),
             colour="red"
  )+
  geom_line(aes(x=real_training_set$YearsExperience,predict(real_regressor,newdata = real_training_set)),
            colour ="blue"
  )+
  ggtitle("Salary vs Exp (Training set)") +
  ylab("Years") +
  xlab("Exp")


#Visualizing training test results
#install.packages('ggplot2')
library(ggplot2)
ggplot() + 
  geom_point(aes(x=real_test_set$YearsExperience,y = real_test_set$Salary),
             colour="red"
  )+
  geom_line(aes(x=real_training_set$YearsExperience,predict(real_regressor,newdata = real_training_set)),
            colour ="blue"
  )+
  ggtitle("Salary vs Exp (Test set)") +
  ylab("Years") +
  xlab("Exp")

File: Part 2 - Simple_Linear_Regression_sample/simple_linear_regression.py
## Simple linear regression

## Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

## Importing the dataset
dataset = pd.read_csv('file:///C:/Pradeep/Project/Learning/ML A-Z/P1 Data Preprocessing/Machine-Learning-A-Z/Part 2 - Regression/Section 4 - Simple Linear Regression/Simple_Linear_Regression_sample/Salary_Data.csv')
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 1].values
print(X)
print(y)



# Splitting the dataset into the Training set and Test set
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 1/3, random_state = 0)


## Fitting the Simple linear regression to the training set . Using LinearRegression Class
from sklearn.linear_model import LinearRegression
regresss=LinearRegression() #no compulsory arguements
regresss.fit(X_train,y_train)

## Predecting the test results (salary of test set)
y_pred=regresss.predict(X_test)

##Visualize the data #Train set observation plot
plt.scatter(X_train,y_train,color='red')
plt.plot(X_train,regresss.predict(X_train),color="blue")
plt.title("Salary vs Experience (Train set)")
plt.xlabel("Year of Salary")
plt.ylabel("Salary")
plt.show()



##Visualize the data #Test set observation plot
plt.scatter(X_test,y_test,color='red')
plt.plot(X_train,regresss.predict(X_train),color="blue")
plt.title("Salary vs Experience (Test Set)")
plt.xlabel("Year of Salary  (Test set)")
plt.ylabel("Salary")
plt.show()

